services/MerchStoreService.ts

```diff
@@ -109,6 +149,79 @@ export default class MerchStoreService {
     });
   }
 
+  /**
+ * Verify that collections have valid photots.
+ */
+  private static verifyCollectionHasValidPhotos(collection: MerchCollection | MerchandiseCollectionModel) {
+    if (collection.collectionPhotos.length > Config.file.MAX_COLLECTION_PHOTO_COUNT) {
+      throw new UserError('Merch items cannot have more than 5 pictures');
+    }
+  }
+
+  /**
+   * Creates a collection photo and assign it the corresponding picture url
+   * and append the photo to the photos list from merchItem
+   * @param collection merch collection uuid
+   * @param properties merch collection photo picture url and position
+   * @returns created collection photo
+  */
+  public async createCollectionPhoto(collection: Uuid, properties: MerchCollectionPhoto):
+  Promise<PublicMerchCollectionPhoto> {
+    return this.transactions.readWrite(async (txn) => {
+      const merchCollection = await Repositories.merchStoreCollection(txn).findByUuid(collection);
+      if (!merchCollection) throw new NotFoundError('Merch item not found');
+
+      // add to the end
+      const position = merchCollection.collectionPhotos.length;
```

Comment:
Ok I think you probably want to position revamp just like what I did to my PR, position should be an input from frontend.